modules/rampart-trust/src/main/java/org/apache/rahas/impl/SAMLTokenIssuer.java

```diff
@@ -167,9 +180,15 @@ public SOAPEnvelope issue(RahasData data) throws TrustException {
 
             String keyType = data.getKeyType();
             SAMLAssertion assertion;
-            if (keyType == null) {
-                throw new TrustException(TrustException.INVALID_REQUEST,
-                        new String[] { "Requested KeyType is missing" });
+            if (StringUtils.isBlank(keyType)) {
+                //According to ws-trust-1.3; <keytype> is an optional URI element.
+                if (StringUtils.isNotBlank(data.getAppliesToAddress())) {
+                    keyType = "http://schemas.xmlsoap.org/ws/2005/05/identity/SymmetricKey";
```

Comment:
Get the appropriate request namespace and build the key type accordingly. Use constants